### PR TITLE
Clean up strl* wrappers

### PR DESCRIPTION
**Pull request by matvore**
strlcat and strlcpy are defined on BSD systems but not GNU/Linux.
Ganesha has its own definitions for these functions, and it is causing
linker errors for me.

strlcat is unused here and in Ganesha, so I propose removing it. I've
renamed strlcpy to avoid conflicts.

### CODE REVIEW

**Comment by dang** [comment]
I think these are supposed to be compatibility functions.  A better solution is to have a FindSTR.cmake that sets HAVE_STRLCAT and HAVE_STRLCPY when they exist.  Are you up on CMake enough to write this?  If not, I can, and you can test it for me.

**Comment by matvore** [comment]
> I think these are supposed to be compatibility functions. A better solution is to have a FindSTR.cmake that sets HAVE_STRLCAT and HAVE_STRLCPY when they exist.

The problem I'm running into is that Ganesha has the same compatibility functions, which generates a linker error. (By extension, anyone depending on ntirpc may be defining similar functions).

Changing cmake logic doesn't address the problem because the problem occurs when building on Linux, where HAVE_STRLCPY is undefined both when building Ganesha and ntirpc.

Concretely, this is the linker error I'm getting:

```
ld: error: backward reference detected: strlcpy in <snip>/libntirpc/clnt_perror.pic.o refers to <snip>/nfs_ganesha/strlcpy.pic.o
```

I tried passing `--warn-backrefs-exclude=*` [0] but this didn't work (maybe my ld is old, but I don't have much control over the build environment I'm using so I can't update it). I'm currently using a workaround where I define `HAVE_STRLCPY` when building NFS Ganesha but not for libntirpc, and turn off the error for implicitly declared functions :/

[0] https://reviews.llvm.org/D77512

**Comment by dang** [comment]
Hmm, this is strange, because I have the opposite problem: if I don't have both versions of strlcpy defined, then I get undefined references.  What is your build chain?  I'm using gcc/gnu-ld (gcc 10.3.1).

**Comment by matvore** [comment]
I'm using Clang, which I believe can warn/error about backrefs while GCC cannot (a search for "backrefs" only revealed pages associated with Clang, not GCC)

TL;DR I must have had something wrong with my build setup because while fixing other things this backrefs error went away. I believe that Ganesha and ntirpc are now using their respective definitions of strlcpy. (backrefs warning occurred when libntirpc's strlcpy was resolving to Ganesha's)

**Comment by dang** [comment]
Good to know, thanks.
